test/e2e/lb_subnets_test.go

```diff
@@ -0,0 +1,419 @@
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.ClassicLoadBalancerParameters.Subnets != nil
+}
+
+// getClusterSubnets populates public and private AWSSubnets objects with a list of
+// the public and private cluster subnets names. Unfortunately these subnets aren't
+// easily accessible in any K8S API objects. However, by using the installer-created subnet
+// naming convention, we can generate the names of the public and private subnets.
+func getClusterSubnets() (public *operatorv1.AWSSubnets, private *operatorv1.AWSSubnets, err error) {
+	// Subnet Naming Convention: <clusterName>-subnet-[public|private]-<availabilityZone>
+	machineSets := &machinev1.MachineSetList{}
+	listOpts := []client.ListOption{
+		client.InNamespace("openshift-machine-api"),
+	}
+	if err := kclient.List(context.Background(), machineSets, listOpts...); err != nil {
+		return nil, nil, fmt.Errorf("failed to get machineSets: %v", err)
+	}
+	// The availability zones can be derived from the providerSpec on MachineSet object.
+	publicSubnets := &operatorv1.AWSSubnets{}
+	privateSubnets := &operatorv1.AWSSubnets{}
+	for _, machineset := range machineSets.Items {
+		providerSpec := new(machinev1.AWSMachineProviderConfig)
+		if err := unmarshalInto(&machineset, providerSpec); err != nil {
+			return nil, nil, fmt.Errorf("failure to unmarshal machineset %q provider spec: %v", machineset.Name, err)
+		}
+
+		if len(providerSpec.Placement.AvailabilityZone) == 0 {
+			return nil, nil, fmt.Errorf("machineset %q availability zone is empty", machineset.Name)
+		}
+		// Build the public and private subnet name according to the installer naming convention.
+		publicSubnet := infraConfig.Status.InfrastructureName + "-subnet-public-" + providerSpec.Placement.AvailabilityZone
+		publicSubnets.Names = append(publicSubnets.Names, operatorv1.AWSSubnetName(publicSubnet))
+		privateSubnet := infraConfig.Status.InfrastructureName + "-subnet-private-" + providerSpec.Placement.AvailabilityZone
+		privateSubnets.Names = append(privateSubnets.Names, operatorv1.AWSSubnetName(privateSubnet))
+	}
+	return publicSubnets, privateSubnets, nil
+}
+
+// unmarshalInto unmarshals a MachineSet's ProviderSpec into the provided providerSpec parameter.
+func unmarshalInto(m *machinev1.MachineSet, providerSpec interface{}) *field.Error {
+	if m.Spec.Template.Spec.ProviderSpec.Value == nil {
+		return field.Required(field.NewPath("providerSpec", "value"), "a value must be provided")
+	}
+
+	if err := json.Unmarshal(m.Spec.Template.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
+		return field.Invalid(field.NewPath("providerSpec", "value"), providerSpec, err.Error())
+	}
+	return nil
+}
+
+// effectuateIngressControllerSubnets manually effectuates updated IngressController subnets by
```

Comment:
> why are manually deleting ?

Because that's something a customer would do after following the condition instructions, and I think should test End-to-End, i.e. make sure that after we observe `LoadBalancerProgressing`, we delete the service, ensure it is populated with the new subnets, and make sure the IngressController status is accurate.

> Should not we just check the progressing status to true and check the message in the IC status which mentions to delete manually or revert back using the patch command ?

We do check for `LoadBalancerProgressing`. I suppose we could try to parse the message to be absolutely certain. I did consider that, but:
1. Felt a little messy/fragile to try to parse the message 
2. We are only changing the subnet, so it's _pretty_ reasonable to assume `LoadBalancerProgressing=true` is because of this subnet update.
3. `TestScopeChange` test doesn't parse the message either, I was modeling off of that E2E test.

So I'm a bit indifferent. I'm open to parsing the message if you feel strongly we should, just want to make sure we are on the same page before refactoring more.

Reply:
OK in original PR I did this : https://github.com/openshift/cluster-ingress-operator/pull/1034/commits/86f94a54e77f46496565b6ec0b38526ab08731e0#diff-160d46f77f47e8e0d35737b019c27a6e4a6f7e7eae4f6e85c0dcdb43b20a2379R317-R365

Reply:
make sure that after we observe LoadBalancerProgressing -- sounds good